realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -857,11 +858,55 @@ private void setModelList(String fieldName, RealmList<DynamicRealmObject> list)
             default:
                 throw new IllegalArgumentException("Unsupported type: " + elementType);
         }
+        final ManagedListOperator<?> operator = getOperator(proxyState.getRealm$realm(), osList, elementType, elementClass);
+
+        if (list.isManaged() && osList.size() == list.size()) {
```

Comment:
When does this happen exactly? Is there a test for it? Is `osList.size == list.size` sufficient to determine that this is the case :question:

Reply:
No, but if the `size()` is different we know with 100% that the lists are not the same, in which case we can use `clear() + insertAll()` which is much faster than manually replacing each element in the the list. 

If they are the same size() we don't know for sure, so we have to revert to manually replacing each element.